lib/pubsub/topic.js

```diff
@@ -178,6 +179,12 @@ Topic.prototype.publish = function(messages, callback) {
     throw new Error('Cannot publish without a message.');
   }
 
+  messages.forEach(function(message) {
+    if (!util.is(message, 'object')) {
```

Comment:
I think wrong objects will still pass the test. It is great if you can make sure the object will very likely be formatted correctly.